java/memory/memory-core/pom.xml

```diff
@@ -37,4 +37,58 @@
     </dependency>
   </dependencies>
 
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-surefire-plugin</artifactId>
+
+        <configuration>
+          <excludes>
+            <!-- Test is only useful when NOT running with add-opens -->
+            <exclude>**/TestOpens.java</exclude>
+          </excludes>
+        </configuration>
+      </plugin>
+    </plugins>
+  </build>
+
+  <profiles>
+    <profile>
+      <id>opens-tests</id>
+      <!-- Run tests WITHOUT add-opens to make sure we fail-fast -->
+      <activation>
+        <jdk>[16,]</jdk>
+      </activation>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-surefire-plugin</artifactId>
+
+            <executions>
+              <execution>
+                <id>opens-tests</id>
+                <phase>test</phase>
+                <goals>
+                  <goal>test</goal>
+                </goals>
+                <configuration>
+                  <!-- Dummy value to stop inheriting the default add-opens flag -->
+                  <argLine>-Dfoo=bar</argLine>
```

Comment:
I could suggest that this line add more arg configuration instead of reset inheriting .values...

Reply:
What do you mean?

Reply:
I mean parent project configure `maven-surefire-plugin` to add `<argLine>--add-opens=java.base/java.nio=ALL-UNNAMED</argLine>`, if the plan is to TestOpens.java fail will be needed to reset Surefire parent configuration

When you run the TestOpens test it's throw the exception?